### PR TITLE
usm: grpc: classification: Extend limits

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Extend our gRPC classification limits

### Motivation
Handle more use-cases of grpc classification

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/ebpf/c/protocols/grpc/helpers.h [hunk 1]
```diff
@@ -8,11 +8,11 @@
...
 #include "protocols/grpc/defs.h"
 
-#define GRPC_MAX_FRAMES_TO_FILTER 10
+#define GRPC_MAX_FRAMES_TO_FILTER 45
 // We only try to process one frame at the moment. Trying to process more yields
 // a verifier issue due to the way clang manages a pointer to the stack.
-#define GRPC_MAX_FRAMES_TO_PROCESS 1
-#define GRPC_MAX_HEADERS_TO_PROCESS 10
+#define GRPC_MAX_FRAMES_TO_PROCESS 10
+#define GRPC_MAX_HEADERS_TO_PROCESS 20
```
Comment (amitslavin) [inline pkg/network/ebpf/c/protocols/grpc/helpers.h:15]: I would expect us to have UTs for the following values in order to make sure that we actually able to process/filter those amounts. Do we have a plan to create those type of tests?

File: pkg/network/ebpf/c/protocols/grpc/helpers.h [hunk 2]
```diff
@@ -178,15 +177,20 @@ static __always_inline grpc_status_t is_grpc(const struct __sk_buff *skb, const
         info.data_off += current_frame.length;
     }
 
+    char content_type_buf[GRPC_CONTENT_TYPE_LEN];
+
 #pragma unroll(GRPC_MAX_FRAMES_TO_PROCESS)
-    for (__u8 i = 0; i < GRPC_MAX_FRAMES_TO_PROCESS && status == PAYLOAD_UNDETERMINED; ++i) {
+    for (__u8 i = 0; i < GRPC_MAX_FRAMES_TO_PROCESS; ++i) {
```
Comment (amitslavin) [inline pkg/network/ebpf/c/protocols/grpc/helpers.h:183]: Does the condition of status == PAYLOAD_UNDETERMINED is not relevant anymore for the iteration pf the frames to process? I mean is it possible for us to be here in a state which is not PAYLOAD_UNDETERMINED, such as PAYLOAD_NOT_GRPC?

**Review by amitslavin — CHANGES_REQUESTED** [review]

**Comment by guyarb** [comment]
rebased on top of main, without changing anything

File: pkg/network/ebpf/c/protocols/grpc/helpers.h (hunk 1 above)
Comment (guyarb) [reply pkg/network/ebpf/c/protocols/grpc/helpers.h:15]: I'll add such tests in the tracer package (where the classification tests sit)

File: pkg/network/ebpf/c/protocols/grpc/helpers.h (hunk 2 above)
Comment (guyarb) [reply pkg/network/ebpf/c/protocols/grpc/helpers.h:183]: no, I just moved the comparison to be later in the process. As you can see, the only place that changes the value of `status` is inside this loop

File: pkg/network/ebpf/c/protocols/grpc/helpers.h (hunk 1 above)
Comment (amitslavin) [reply pkg/network/ebpf/c/protocols/grpc/helpers.h:15]: In a sperate PR?

**Review by amitslavin — APPROVED** [review]

File: pkg/network/ebpf/c/protocols/grpc/helpers.h (hunk 1 above)
Comment (guyarb) [reply pkg/network/ebpf/c/protocols/grpc/helpers.h:15]: no, in that PR

**Pull request closed without merging**